receiver/sqlserverreceiver/scraper_test.go

```diff
@@ -47,13 +50,17 @@ func TestSuccessfulScrape(t *testing.T) {
 
 	// Ensure all metrics are received when all are enabled.
 	cfg.MetricsBuilderConfig.Metrics.SqlserverDatabaseIoReadLatency.Enabled = true
+	cfg.MetricsBuilderConfig.Metrics.SqlserverResourcePoolDiskThrottledReadRate.Enabled = true
+	cfg.MetricsBuilderConfig.Metrics.SqlserverResourcePoolDiskThrottledWriteRate.Enabled = true
+	cfg.MetricsBuilderConfig.Metrics.SqlserverProcessesBlocked.Enabled = true
+
 	scrapers := setupSQLServerScrapers(receivertest.NewNopCreateSettings(), cfg)
 	assert.NotNil(t, scrapers)
 
 	for _, scraper := range scrapers {
 		err := scraper.Start(context.Background(), componenttest.NewNopHost())
 		assert.NoError(t, err)
-		defer func() { assert.NoError(t, scraper.Shutdown(context.Background())) }()
```

Comment:
The anonymous `func()` has to be removed here, as with it, `scraper.Shutdown` is called twice on the second of two scrapers, and not called on the first scraper. (This is how defer works with anonymous funcs by definition, we just need to use them properly for the updated use case.)